cmd/maya-apiserver/spc-watcher/select_disk.go

```diff
@@ -137,28 +144,45 @@ func diskSelector(nodeDiskMap map[string]*diskList, poolType string) *nodeDisk {
 		},
 	}
 
-	// requiredDiskCount will hold the required number of disk that should be selected from a qualified
+	// diskCount will hold the number of disk that will be selected from a qualified
 	// node for specific pool type
-	var requiredDiskCount int
-	// If pool type is striped, 1 disk should be selected
+	var diskCount int
+	// minRequiredDiskCount will hold the required number of disk that should be selected from a qualified
+	// node for specific pool type
+	var minRequiredDiskCount int
+	// If pool type is striped, at least 1 disk should be selected
 	if poolType == string(v1alpha1.PoolTypeStripedCPV) {
-		requiredDiskCount = int(v1alpha1.StripedDiskCountCPV)
+		minRequiredDiskCount = int(v1alpha1.StripedDiskCountCPV)
 	}
-	// If pool type is mirrored, 2 disks should be selected
+	// If pool type is mirrored, at least 2 disks should be selected
 	if poolType == string(v1alpha1.PoolTypeMirroredCPV) {
-		requiredDiskCount = int(v1alpha1.MirroredDiskCountCPV)
+		minRequiredDiskCount = int(v1alpha1.MirroredDiskCountCPV)
 	}
 	// Range over the nodeDiskMap map to get the list of disks
 	for node, val := range nodeDiskMap {
 
 		// If the current disk count on the node is less than the required disks
 		// then this is a dirty node and it will not qualify.
-		if len(val.items) < requiredDiskCount {
+		if len(val.items) < minRequiredDiskCount {
 			continue
 		}
+		if provisioningType == ProvisioningTypeManual {
+			diskCount = len(val.items)
+		}
+		if provisioningType == ProvisioningTypeAuto {
+			diskCount = minRequiredDiskCount
+		}
 		// Select the required disk from qualified nodes.
-		for i := 0; i < requiredDiskCount; i++ {
-			selectedDisk.disks.items = append(selectedDisk.disks.items, val.items[i])
+		if poolType == string(v1alpha1.PoolTypeStripedCPV) {
+			for i := 0; i < diskCount; i++ {
+				selectedDisk.disks.items = append(selectedDisk.disks.items, val.items[i])
+			}
+		}
+		if poolType == string(v1alpha1.PoolTypeMirroredCPV) {
+			for i := 0; i < diskCount/2*2; i = i + 2 {
```

Comment:
Please check if the logic can be made easy to understand. I am not able to understand the logic. Are we dividing by 2 & then multiplying by 2?

Reply:
Discussed with you on this.